src/click/types.py

```diff
@@ -836,11 +836,14 @@ def convert(
 
         if not is_dash:
             if self.resolve_path:
-                # realpath on Windows Python < 3.8 doesn't resolve symlinks
+                # get the absolute directory path of "rv"
+                dir_ = os.path.dirname(os.path.abspath(rv))
                 if os.path.islink(rv):
                     rv = os.readlink(rv)
 
-                rv = os.path.realpath(rv)
+                # Prepends dir_ to resolved relative symlink
+                # Has no effect if passed an absolute path
+                rv = os.path.join(dir_, rv)
```

Comment:
It is used quite a lot in the codebase. I think anyone working on it would become accustomed to it quite quickly. Kind of like `retval` in some repos.